rfc/user-secrets.md

```diff
@@ -0,0 +1,186 @@
+These secrets provide authorization metadata for integration with Fiat so that multiple tenants can share the same Spinnaker environment.
+
+### Goals and Non-Goals
+
+* Goals
+  - Enable Spinnaker to use external secrets managers through a common Spinnaker identity which secrets owners can grant read access.
+  - Enable Spinnaker to enforce access controls to these secrets based on authorization data delivered with the secret.
+  - Enable Clouddriver account credentials to use authorized user secrets.
+  - Enable other Spinnaker services to use user secrets via common Kork library.
+* Non-Goals
+  - Direct storage of secrets (best done through external secrets managers).
+  - Modification of user secrets contents.
+  - Management of user secrets.
+
+## Motivation and Rationale
+
+Clouddriver has an alpha feature for managing accounts backed by a SQL database.
+Using the existing Kork `SecretEngine` API, account definitions can reference secrets through `encrypted:...` and `encryptedFile:...` URIs which are resolved at load time.
+These engines rely on the Spinnaker runtime identity to make API calls to their respective services to retrieve the referenced secrets.
+This functionality was originally used in Halyard for resolving secrets at deploy time where there is only one relevant identity to authorize.
+Extending this secrets manager integration to enable multiple tenants requires additional authorization details as the access controls of the secrets manager are used for granting read access to Spinnaker itself.
+Without the authorization details, any user could use one another's secrets if the same encrypted URI is used which is not secure in a multi-tenant setting.
+These authorization details are included with other metadata and the secret data in the secret itself, so an encoding format is defined.
+
+Early stakeholders for this effort are:
+
+* Apple
+
+## Timeline
+
+The alpha version of this feature is targeted for 2022Q2 and is being contributed by Apple.
+Development began in Q1 coinciding with the related Clouddriver Account Management API feature which initially integrated with the Kubernetes V2 provider.
+User secrets will be integrated into this API along with updates to existing Kork `SecretEngine` for AWS Secrets Manager for a reference implementation.
+
+## Design
+
+User secrets implement `UserSecret` which provides some common operations around extracting secret data and checking permissions.
+Implementations provide a `type` property which determines the structure of the secret data.
+These secrets can be referenced through a `secret://` URI using the `UserSecretReference` class to parse.
+Each URI is structured as `secret://secret-engine-id?param1=value1&param2=value2&param3=value3` where _secret-engine-id_ is the `SecretEngine` identifier and the parameter key/value pairs are specific to each engine.
```

Comment:
I'm wondering if we could switch to THIS format for secrets vs. the encrypted: syntax to standardize this a bit more.  Overall I like this format :) More sense as an official URI structure!